src/compiler/checker.ts

```diff
@@ -21460,6 +21460,10 @@ namespace ts {
                     const fileSymbol = getSymbolOfNode(container);
                     return fileSymbol && getTypeOfSymbol(fileSymbol);
                 }
+                else if (container.externalModuleIndicator) {
+                    // TODO: Maybe issue a better error than 'object is possibly undefined'

```

Comment:
I do think this error is a little confusing if the type is unambiguously `undefined`, but I’m just realizing that that’s the case for any usage of `undefined`, so I guess it’s fine.